### ISSUE TITLE
TypeScript generator doesn't resolve Rust Result

### ISSUE THREAD

**Issue by NuroDev** [1Password/typeshare#5]
## What

Currently in the process of testing things out a bit & seeing what does & doesn't work & while `Option`'s seem to work fine, it seem a Rust `Result` seems to get passed straight through rather than being resolved to something else.

### Example:

```rs
// input.rs

#[typeshare]
struct Example {
    foo: Result<u32, String>,
    bar: Option<u32>,
}
```

```bash
typeshare-cli -l typescript -o ./index.ts input.rs
```

```ts
/*
 Generated by typeshare 1.0.0
*/

export interface Example {
	foo: Result<number, string>; // Cannot find name 'Result'.ts(2304)
	bar?: number;
}


```

## How

I'm unsure if there is already a better plan to handle resolving these, whether that just be unwrapping the resulting type & returning that or adding a `Result` type to the generated types similar something like [ts-results's `Result` type?](https://github.com/vultix/ts-results/blob/master/src/result.ts#L282)

**Comment by InquisitivePenguin** [comment]
For right now, unwrapping the resulting type would probably be your best bet. We don't have a cross-language type definition to wrap `Result` at the moment, but if you think it's useful this is definitely something we could explore.

**Comment by NuroDev** [comment]
I agree, at least for now, defaulting to unwrapping the success type.
Then, maybe in the future, offer some kind of config option, likely in the `typeshare.toml` (Not sure how this would work on a per-language basis), to include / append a basic `Result` type like what I linked above from `ts-result`

**Comment by Brendonovich** [comment]
Over in [rspc](https://github.com/oscartbeaumont/rspc) we just ignore the error type of results, since that's how serde treats them. At least in TypeScript, errors get thrown and aren't usually part of a function's return value.
